streams/src/main/java/org/apache/kafka/streams/internals/AutoOffsetResetInternal.java

```diff
@@ -31,6 +31,13 @@ public StrategyType offsetResetStrategy() {
         return offsetResetStrategy;
     }
 
+    // calling `Optional.get()` w/o checking `isPresent()` results in a warning that we suppress
+    // (there is unfortunately no speficif suppression for `Optional.get()` warnings)
+    //
+    // we do call `get()` unconditionally on purpose here, as it should only be called before `offsetResetStrategy()`
+    // was used to verify that the strategy is "by_duration"
+    // if the strategy is not "by_duration" and `duration()` is called,
+    // we want to fail right away to surface the bug in the caller code
```

Comment:
actually now I'm thinking we should maybe just have a check instead of suppressing the warnings -- I know this _should_ never happen, but if due to some bug (eg introduced in a future refactoring) it actually is Optional.empty(), then it will be better to check and throw a helpful error message here. it's not a heavy API call so we may as well?

Reply:
eg 
```
if (isEmpty) {
  final String errorMsg = String.format("Duration was unexpectedly empty. This should only ever be called after checking that the offsetResetStrategy() was 'by_duration'. Actual strategy was %s", offsetResetStrategy());
  log.error(errorMsg);
  throw new IllegalStateException(errorMsg);
}
```

Reply:
Specifically it will be helpful to say in the error message that this should always be present when the strategy is duration, and then log the actual strategy so we can tell if the bug was using this API on another strategy or a failure to set the duration. Just trying to imagine what would be helpful to know if this ever showed up in a JIRA ticket 🤷‍♀️

Reply:
Works for me -- I did consider it at some point, but thought just hitting a `NoSuchElementException` would be good enough... Will update.